### ISSUE TITLE
runtime: fatal error: unexpected signal during runtime execution

### ISSUE THREAD

**Issue by xiang90** [golang/go#13046]
Go version: 1.5
Arch: linux/amd64. kernel 4.1.7

stack trace
```
fatal error: unexpected signal during runtime execution
[signal 0xb code=0x80 addr=0x0 pc=0x40f024]

goroutine 2366508 [running]:
runtime.throw(0xe5a5e0, 0x2a)
    /usr/local/go/src/runtime/panic.go:527 +0x90 fp=0xc8215a68e8 sp=0xc8215a68d0
runtime.sigpanic()
    /usr/local/go/src/runtime/sigpanic_unix.go:12 +0x5a fp=0xc8215a6938 sp=0xc8215a68e8
runtime.mallocgc(0x30, 0xa88aa0, 0x1, 0xc8220aac30)
    /usr/local/go/src/runtime/malloc.go:617 +0x884 fp=0xc8215a6a08 sp=0xc8215a6938
runtime.newarray(0xa88aa0, 0x5, 0x7f240d)
    /usr/local/go/src/runtime/malloc.go:777 +0xc9 fp=0xc8215a6a48 sp=0xc8215a6a08
runtime.makeslice(0xd21920, 0x5, 0x5, 0x0, 0x0, 0x0)
    /usr/local/go/src/runtime/slice.go:32 +0x165 fp=0xc8215a6a98 sp=0xc8215a6a48
math/big.nat.divLarge(0xc8216a00c0, 0x1, 0x5, 0xc8220aac30, 0x6, 0xa, 0xc8220aab90, 0x5, 0xa, 0xc8220aabe0, ...)
    /usr/local/go/src/math/big/nat.go:578 +0x4d6 fp=0xc8215a6c48 sp=0xc8215a6a98
math/big.nat.div(0xc8216a00c0, 0x1, 0x5, 0x0, 0x0, 0x0, 0xc8220aab90, 0x5, 0xa, 0xc8220aabe0, ...)
    /usr/local/go/src/math/big/nat.go:545 +0x2f5 fp=0xc8215a6cf0 sp=0xc8215a6c48
math/big.(*Int).QuoRem(0xc8215a6e48, 0xc8217115e0, 0xc821711600, 0xc821711620, 0xc8215a6e48, 0xc821711600)
    /usr/local/go/src/math/big/int.go:227 +0xe6 fp=0xc8215a6db8 sp=0xc8215a6cf0
math/big.(*Int).GCD(0xc8215a6f60, 0xc8215a6fe0, 0x0, 0xc821710d40, 0xc82011b540, 0x10)
    /usr/local/go/src/math/big/int.go:464 +0x2c6 fp=0xc8215a6f30 sp=0xc8215a6db8
math/big.(*Int).ModInverse(0xc8215a6fe0, 0xc821710d40, 0xc82011b540, 0x7ea700)
    /usr/local/go/src/math/big/int.go:578 +0x61 fp=0xc8215a6f88 sp=0xc8215a6f30
crypto/elliptic.(*CurveParams).affineFromJacobian(0xc820202240, 0xc821710d00, 0xc821710d20, 0xc821710d40, 0xc821710cc0, 0xc821710ca0)
    /usr/local/go/src/crypto/elliptic/elliptic.go:91 +0xb6 fp=0xc8215a7008 sp=0xc8215a6f88
crypto/elliptic.(*CurveParams).ScalarMult(0xc820202240, 0xc82011b600, 0xc82011b640, 0xc8215baae0, 0x30, 0x30, 0x30, 0x30)
    /usr/local/go/src/crypto/elliptic/elliptic.go:265 +0x259 fp=0xc8215a7128 sp=0xc8215a7008
crypto/elliptic.(*CurveParams).ScalarBaseMult(0xc820202240, 0xc8215baae0, 0x30, 0x30, 0xc8215b7aa0, 0x0)
    /usr/local/go/src/crypto/elliptic/elliptic.go:269 +0x51 fp=0xc8215a7170 sp=0xc8215a7128
crypto/ecdsa.Sign(0x7ffa7ce9c358, 0xc820012720, 0xc8202051d0, 0xc8215b7a40, 0x20, 0x20, 0x0, 0x0, 0x0, 0x0)
    /usr/local/go/src/crypto/ecdsa/ecdsa.go:183 +0x5d8 fp=0xc8215a72e0 sp=0xc8215a7170
crypto/ecdsa.(*PrivateKey).Sign(0xc8202051d0, 0x7ffa7ce9c358, 0xc820012720, 0xc8215b7a40, 0x20, 0x20, 0x7ffa7ce5f6b8, 0xc8214d5790, 0x0, 0x0, ...)
    /usr/local/go/src/crypto/ecdsa/ecdsa.go:60 +0x92 fp=0xc8215a7358 sp=0xc8215a72e0
crypto/tls.(*clientHandshakeState).doFullHandshake(0xc8215a7e48, 0x0, 0x0)
    /usr/local/go/src/crypto/tls/handshake_client.go:466 +0x19f0 fp=0xc8215a7a10 sp=0xc8215a7358
crypto/tls.(*Conn).clientHandshake(0xc821252840, 0x0, 0x0)
    /usr/local/go/src/crypto/tls/handshake_client.go:206 +0x164e fp=0xc8215a7ef8 sp=0xc8215a7a10
crypto/tls.(*Conn).Handshake(0xc821252840, 0x0, 0x0)
    /usr/local/go/src/crypto/tls/conn.go:974 +0xec fp=0xc8215a7f38 sp=0xc8215a7ef8
net/http.(*Transport).dialConn.func3(0xc821252840, 0xc821b7cd00, 0xc821482120)
    /usr/local/go/src/net/http/transport.go:662 +0x21 fp=0xc8215a7f78 sp=0xc8215a7f38
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:1696 +0x1 fp=0xc8215a7f80 sp=0xc8215a7f78
created by net/http.(*Transport).dialConn
    /usr/local/go/src/net/http/transport.go:667 +0x7bc
```

**Comment by RLH** [comment]
Is this reproducible? If so how?

**Comment by xiang90** [comment]
@RLH Sorry. This is a panic that a user reported to me. We tried to reproduce it but failed. And it seems to happen very infrequently. 

I suspect this is a go runtime bug. If you think this is a user program bug, I can try to check the related code. 

Also I am happy to look into the related code inside go if you are sure this is a go runtime issue.

Let me know if you want more information.

**Comment by minux** [comment]
fatal error: unexpected signal during runtime execution
[signal 0xb code=0x80 addr=0x0 pc=0x40f024]

The code is interesting. Normally, it should 0x1 (SEGV_MAPERR).
(The other possibilities are 2 for SEGV_ACCERR, and 3 for
SEGV_BNDERR)

code 0x80 just signifies that it's coming from kernel, but doesn't
give a reason for the signal.

And looking at the code:
609             if v.ptr() == nil {
610                 systemstack(func() {
611                     mCache_Refill(c, int32(sizeclass))
612                 })
613                 shouldhelpgc = true
614                 s = c.alloc[sizeclass]
615                 v = s.freelist
616             }
617             s.freelist = v.ptr().next

v.ptr() shouldn't be nil at line 617 unless someone else is also
modifying it. (but it's from gomcache, nobody else should be
modifying it concurrently.

Could you tell us more about the program? Does it use cgo or
unsafe? Have you tried to run it with race detector enabled?

**Comment by davecheney** [comment]
If this is coming from the kernel, there will be a report in dmesg (there is on arm64 systems)

**Comment by xiang90** [comment]
> Could you tell us more about the program? Does it use cgo or
unsafe? 

The program is etcd (https://github.com/coreos/etcd). The core part of etcd does not involve cgo/unsafe, but we are using external pkgs. So I am not sure. 

> Have you tried to run it with race detector enabled?

A normal go user program can corrupt runtime data? Or the race detector can detect data race in cgo? I cannot reproduce the issue myself right now. I do not know if it helps to collect data by enabling race detector.

**Comment by davecheney** [comment]
> A normal go user program can corrupt runtime data? 

A program with a data race is undefined, it can do _anything_ including corrupting runtime data structures.

**Comment by davecheney** [comment]
@xiang90 i had a quick play with coreos/etcd and found several packages fail the race detector. Are these know failures ? I think you should address those in coreos/etcd before we can investigate.

**Comment by lorenz** [comment]
I encountered the issue and checked dmesg, nothing was printed there.

**Comment by aclements** [comment]
@lorenz, can you post your panic message and traceback?

**Comment by lorenz** [comment]
Is in [here](https://github.com/coreos/etcd/issues/3745).

**Comment by aclements** [comment]
Ah, sorry, I hadn't realized you were the poster of the original bug.

I'm inclined to agree with @davecheney that there's not much we can do until the races are fixed.

**Comment by RLH** [comment]
I agree with Austin and Dave that this should be closed until the program
passes the race detector. That said I am sympathetic to Xlang's
expectations that the runtime needs to be more robust against racy code.
There is a history of C malloc implementations pulling their free lists
from the heap and building them outside of the heap to avoid triggering
heap corruption bugs. Obviously racy programs will always be a problem and
we won't be able to completely insulate the runtime from them. Long term we
are headed in this direction.

**Comment by xiang90** [comment]
@RLH 

After reading runtime malloc code, I agree with @davecheney that the user code can potentially corrupt runtime code. But to be honest, I still feel the possibility is still too low that a newly allocated freelist is corrupted. 

I would like to close this issue on go upstream for now. I will reopen it if I find something.

**Comment by xiang90** [comment]
@davecheney Just want to make sure we are fixing the current thing (at least fixing what you have seen)

>  i had a quick play with coreos/etcd and found several packages fail the race detector.

How did you play with coreos/etcd? How did you find out the races?

**Comment by davecheney** [comment]
> How did you play with coreos/etcd? How did you find out the races?
```
go get github.com/coreos/etcd
go test -race github.com/coreos/etcd/...
```

**Comment by fusion809** [comment]
Same issue here when running `docker build .` on 64-bit Arch Linux. Here is the exact error:

```bash
fatal error: unexpected signal during runtime execution
[signal 0xb code=0x1 addr=0x0 pc=0x0]

runtime stack:
runtime.throw(0x196e440, 0x2a)
        /usr/lib/go/src/runtime/panic.go:547 +0x90
runtime.sigpanic()
        /usr/lib/go/src/runtime/sigpanic_unix.go:12 +0x5a
runtime.systemstack(0xc820022000)
        /usr/lib/go/src/runtime/asm_amd64.s:295 +0x79
runtime.mstart()
        /usr/lib/go/src/runtime/proc.go:1051

goroutine 8 [running]:
runtime.systemstack_switch()
        /usr/lib/go/src/runtime/asm_amd64.s:245 fp=0xc82002ceb0 sp=0xc82002cea8
runtime.writebarrierptr_nostore1(0xc8204ba300, 0xc82002cfa0)
        /usr/lib/go/src/runtime/mbarrier.go:121 +0x71 fp=0xc82002cee8 sp=0xc82002ceb0
runtime.writebarrierptr(0xc8204ba300, 0xc82002cfa0)
        /usr/lib/go/src/runtime/mbarrier.go:143 +0x72 fp=0xc82002cf18 sp=0xc82002cee8
runtime.gopark(0xc8202cc3b8, 0xc82002cfa0, 0x18616a0, 0x10, 0x14, 0x0)
        /usr/lib/go/src/runtime/proc.go:255 +0x1b8 fp=0xc82002cf40 sp=0xc82002cf18
runtime.gcBgMarkWorker(0xc82001a000)
        /usr/lib/go/src/runtime/mgc.go:1403 +0xda fp=0xc82002cfb8 sp=0xc82002cf40
runtime.goexit()
        /usr/lib/go/src/runtime/asm_amd64.s:1998 +0x1 fp=0xc82002cfc0 sp=0xc82002cfb8
created by runtime.gcBgMarkStartWorkers
        /usr/lib/go/src/runtime/mgc.go:1330 +0x92

goroutine 1 [runnable]:
strings.Join(0xc820251ce8, 0x2, 0x2, 0x175b6f8, 0x1, 0x0, 0x0)
        /usr/lib/go/src/strings/strings.go:369 +0x3c3
path/filepath.join(0xc820251ce8, 0x2, 0x2, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path_unix.go:42 +0xfa
path/filepath.Join(0xc820251ce8, 0x2, 0x2, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:206 +0x41
path/filepath.walk(0xc820523d40, 0xcc, 0x7fd82ab46468, 0xc820523e10, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:367 +0x37a
path/filepath.walk(0xc8205225b0, 0xc4, 0x7fd82ab46468, 0xc820522680, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc82017e000, 0xc2, 0x7fd82ab46468, 0xc82017e0d0, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc8203dcc00, 0xba, 0x7fd82ab46468, 0xc820454d00, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc8203dc840, 0xb2, 0x7fd82ab46468, 0xc820454c30, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc8203efe40, 0xa6, 0x7fd82ab46468, 0xc8203ed110, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc820156640, 0x9c, 0x7fd82ab46468, 0xc820555ad0, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc82031a1e0, 0x96, 0x7fd82ab46468, 0xc820133110, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc8203de3f0, 0x8e, 0x7fd82ab46468, 0xc820133040, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc8203de900, 0x89, 0x7fd82ab46468, 0xc82017c1a0, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc82042de00, 0x7f, 0x7fd82ab46468, 0xc820425380, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc82026fc00, 0x72, 0x7fd82ab46468, 0xc8202201a0, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc82039bf20, 0x5d, 0x7fd82ab46468, 0xc820163ee0, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc8203ac5a0, 0x50, 0x7fd82ab46468, 0xc820163c70, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc8203ac460, 0x4c, 0x7fd82ab46468, 0xc820163040, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc8203ac320, 0x41, 0x7fd82ab46468, 0xc820162f70, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc8203bcd40, 0x3d, 0x7fd82ab46468, 0xc8203f6340, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.walk(0xc82038dc40, 0x31, 0x7fd82ab46468, 0xc8203b1930, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:374 +0x4fc
path/filepath.Walk(0xc82038dc40, 0x31, 0xc8203b2060, 0x0, 0x0)
        /usr/lib/go/src/path/filepath/path.go:396 +0xe1
github.com/docker/docker/builder.ValidateContextDirectory(0xc82038d800, 0x31, 0x0, 0x0, 0x0, 0x0, 0x0)
        /build/docker/src/docker/.gopath/src/github.com/docker/docker/builder/context.go:68 +0x10b
github.com/docker/docker/api/client.(*DockerCli).CmdBuild(0xc82026fa80, 0xc82000a440, 0x1, 0x1, 0x0, 0x0)
        /build/docker/src/docker/.gopath/src/github.com/docker/docker/api/client/build.go:144 +0x1e72
reflect.callMethod(0xc820389e30, 0xc820253c58)
        /usr/lib/go/src/reflect/value.go:631 +0x1fc
reflect.methodValueCall(0xc82000a440, 0x1, 0x1, 0x1, 0xc820389e30, 0x0, 0x0, 0xc820389e30, 0x0, 0x47a3c4, ...)
        /usr/lib/go/src/reflect/asm_amd64.s:29 +0x36
github.com/docker/docker/cli.(*Cli).Run(0xc820389620, 0xc82000a430, 0x2, 0x2, 0x0, 0x0)
        /build/docker/src/docker/.gopath/src/github.com/docker/docker/cli/cli.go:89 +0x384
main.main()
        /build/docker/src/docker/docker/docker.go:63 +0x435

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/lib/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 6 [syscall]:
os/signal.signal_recv(0x0)
        /usr/lib/go/src/runtime/sigqueue.go:116 +0x132
os/signal.loop()
        /usr/lib/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
        /usr/lib/go/src/os/signal/signal_unix.go:28 +0x37
```

**Comment by aclements** [comment]
@fusion809, please open a new issue. Thanks!

**Comment by vkg70** [comment]
I'm getting this error when I run the cf push command. 

unexpected fault address 0x2a20eeb8ee7a
fatal error: fault
[signal 0xb code=0x1 addr=0x2a20eeb8ee7a pc=0x2a20eeb8ee7a]

goroutine 405 [running]:
runtime.throw(0x871810, 0x5)
	/usr/local/go/src/runtime/panic.go:527 +0x90 fp=0xc820041928 sp=0xc820041910
runtime.sigpanic()
	/usr/local/go/src/runtime/sigpanic_unix.go:27 +0x2ba fp=0xc820041978 sp=0xc820041928
net/http.(*Transport).dial(0xc8204a8630, 0x875228, 0x3, 0xc820326b40, 0x16, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:499 +0x79 fp=0xc8200419c0 sp=0xc820041978
net/http.(*Transport).dialConn(0xc8204a8630, 0x0, 0xc8204a85a0, 0x5, 0xc820326b40, 0x16, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:596 +0x19a9 fp=0xc820041f00 sp=0xc8200419c0
net/http.(*Transport).getConn.func4(0xc8204a8630, 0x0, 0xc8204a85a0, 0x5, 0xc820326b40, 0x16, 0xc82035d740)
	/usr/local/go/src/net/http/transport.go:549 +0x66 fp=0xc820041f78 sp=0xc820041f00
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1 fp=0xc820041f80 sp=0xc820041f78
created by net/http.(*Transport).getConn
	/usr/local/go/src/net/http/transport.go:551 +0x265

goroutine 1 [select]:
net/http.(*Transport).getConn(0xc8204a8630, 0xc82040f500, 0x0, 0xc8204a85a0, 0x5, 0xc820326b40, 0x16, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:554 +0x663
net/http.(*Transport).RoundTrip(0xc8204a8630, 0xc82040f500, 0xb4, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:228 +0x4ac
net/http.send(0xc82040f500, 0x19644f8, 0xc8204a8630, 0xc8204a9b90, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:220 +0x52c
net/http.(*Client).send(0xc8203e8b10, 0xc82040f500, 0x82, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:143 +0x15a
net/http.(*Client).doFollowingRedirects(0xc8203e8b10, 0xc82040f500, 0x9f24c0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:380 +0xbc0
net/http.(*Client).Do(0xc8203e8b10, 0xc82040f500, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:175 +0x9c
github.com/cloudfoundry/cli/cf/net.Gateway.doRequest(0x1c02bb8, 0xc8204a87e0, 0x9f1aa0, 0x1, 0x12a05f200, 0x0, 0x0, 0x0, 0x1c02950, 0xc8204e3b60, ...)
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/tmp/cli_gopath/src/github.com/cloudfoundry/cli/cf/net/gateway.go:425 +0xe6
github.com/cloudfoundry/cli/cf/net.Gateway.doRequestAndHandlerError(0x1c02bb8, 0xc8204a87e0, 0x9f1aa0, 0x1, 0x12a05f200, 0x0, 0x0, 0x0, 0x1c02950, 0xc8204e3b60, ...)
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/tmp/cli_gopath/src/github.com/cloudfoundry/cli/cf/net/gateway.go:399 +0x5b
github.com/cloudfoundry/cli/cf/net.Gateway.doRequestHandlingAuth(0x1c02bb8, 0xc8204a87e0, 0x9f1aa0, 0x1, 0x12a05f200, 0x0, 0x0, 0x0, 0x1c02950, 0xc8204e3b60, ...)
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/tmp/cli_gopath/src/github.com/cloudfoundry/cli/cf/net/gateway.go:370 +0x18d
github.com/cloudfoundry/cli/cf/net.Gateway.performRequestForResponseBytes(0x1c02bb8, 0xc8204a87e0, 0x9f1aa0, 0x1, 0x12a05f200, 0x0, 0x0, 0x0, 0x1c02950, 0xc8204e3b60, ...)
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/tmp/cli_gopath/src/github.com/cloudfoundry/cli/cf/net/gateway.go:249 +0xc9
github.com/cloudfoundry/cli/cf/net.Gateway.PerformRequestForJSONResponse(0x1c02bb8, 0xc8204a87e0, 0x9f1aa0, 0x1, 0x12a05f200, 0x0, 0x0, 0x0, 0x1c02950, 0xc8204e3b60, ...)
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/tmp/cli_gopath/src/github.com/cloudfoundry/cli/cf/net/gateway.go:271 +0x97
github.com/cloudfoundry/cli/cf/net.Gateway.GetResource(0x1c02bb8, 0xc8204a87e0, 0x9f1aa0, 0x1, 0x12a05f200, 0x0, 0x0, 0x0, 0x1c02950, 0xc8204e3b60, ...)
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/tmp/cli_gopath/src/github.com/cloudfoundry/cli/cf/net/gateway.go:99 +0x156
github.com/cloudfoundry/cli/cf/api/applications.CloudControllerApplicationRepository.ReadFromSpace(0x1c02950, 0xc8204e3b60, 0x1c02bb8, 0xc8204a87e0, 0x9f1aa0, 0x1, 0x12a05f200, 0x0, 0x0, 0x0, ...)
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/tmp/cli_gopath/src/github.com/cloudfoundry/cli/cf/api/applications/applications.go:77 +0x383
github.com/cloudfoundry/cli/cf/api/applications.CloudControllerApplicationRepository.Read(0x1c02950, 0xc8204e3b60, 0x1c02bb8, 0xc8204a87e0, 0x9f1aa0, 0x1, 0x12a05f200, 0x0, 0x0, 0x0, ...)
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/tmp/cli_gopath/src/github.com/cloudfoundry/cli/cf/api/applications/applications.go:71 +0x11c
github.com/cloudfoundry/cli/cf/api/applications.(*CloudControllerApplicationRepository).Read(0xc8204a8900, 0xc820449300, 0x11, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	<autogenerated>:3 +0x104
github.com/cloudfoundry/cli/cf/commands/application.(*Push).createOrUpdateApp(0xc820456400, 0xc820444500, 0x0, 0xc820444580, 0xc820448000, 0xc820446418, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/tmp/cli_gopath/src/github.com/cloudfoundry/cli/cf/commands/application/push.go:334 +0x143
github.com/cloudfoundry/cli/cf/commands/application.(*Push).Execute(0xc820456400, 0xeae5e0, 0xc82046eb40)
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/tmp/cli_gopath/src/github.com/cloudfoundry/cli/cf/commands/application/push.go:152 +0x326
main.main()
	/Users/cffadmin/go-agent/pipelines/Mac-OSX-Unit-Tests/src/github.com/cloudfoundry/cli/main/main.go:89 +0x178e

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 19 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 320 [select]:
net/http.(*persistConn).writeLoop(0xc8200d26e0)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 319 [IO wait]:
net.runtime_pollWait(0x19a5740, 0x72, 0xc8200700a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8200ac140, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8200ac140, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8200ac0e0, 0xc8203b4400, 0x400, 0x400, 0x0, 0x1960028, 0xc8200700a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8203506d0, 0xc8203b4400, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc8204c8210, 0x1c05420, 0xc8203506d0, 0x5, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc8204b82c0, 0x9f2917, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc8204b82c0, 0xc8203a1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x1969dd8, 0xc8204b82c0, 0xc8200d2738, 0xc8203a1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc8201e4380, 0xc8203a1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc82006c780)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc82006c780, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8200d26e0)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

**Comment by bradfitz** [comment]
@vkg70, this bug is old and closed. Please file a new bug if you have a problem, and include all the details we might need to debug.
